### PR TITLE
bpo-44808: fixes test for interactive inspect getsource of a class

### PR DESCRIPTION
**Pull request by akulakov**
Fixes unsafe test that was added in #27171 

https://bugs.python.org/issue44808

### CODE REVIEW

File: Lib/test/test_inspect.py [hunk 1]
```diff
@@ -586,19 +586,13 @@ def test_getsource_on_code_object(self):
...
-        self.main = sys.modules['__main__']
-        self.mod = mod.ParrotDroppings.__module__
-        class MockModule:
-            __file__ = None
-        sys.modules['__main__'] = MockModule
-        mod.ParrotDroppings.__module__ = '__main__'
-        with self.assertRaisesRegex(OSError, 'source code not available') as e:
-            inspect.getsource(mod.ParrotDroppings)
+        code = "import sys, inspect; \
+                sys.modules['__main__'].__file__ = None; \
```
Comment (ambv) [inline Lib/test/test_inspect.py]:
```suggestion
        # bpo-44648: simulate a REPL session;
        # there is no `__file__` in the __main__ module
        code = "import sys, inspect; \
                assert not hasattr(sys.modules['__main__'], '__file__'); \
```

**Review by ambv — APPROVED** [review]
This is a very elegant improvement. Consider making it more like the REPL by *not having* a `__file__` attribute at all.

File: Lib/test/test_inspect.py (hunk 1 above)
Comment (akulakov) [reply Lib/test/test_inspect.py]: Thanks! This makes sense.

**Comment by akulakov** [comment]
The test failure seems to be unrelated asyncio test:

>  FAIL: test_sendfile_close_peer_in_the_middle_of_receiving (test.test_asyncio.test_sendfile.ProactorEventLoopTests)
2021-08-03T11:55:01.0792071Z ----------------------------------------------------------------------
2021-08-03T11:55:01.0792702Z Traceback (most recent call last):
2021-08-03T11:55:01.0793490Z   File "D:\a\1\s\lib\test\test_asyncio\test_sendfile.py", line 457, in test_sendfile_close_peer_in_the_middle_of_receiving
2021-08-03T11:55:01.0794230Z     with self.assertRaises(ConnectionError):
2021-08-03T11:55:01.0794943Z     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2021-08-03T11:55:01.0795542Z AssertionError: ConnectionError not raised

**Comment by ambv** [comment]
Oops, we don't want a 3.9 backport actually.

**Pull request merged**